### PR TITLE
Fixing issues found in #70 (companion object checks)

### PR DESCRIPTION
**Pull request by akuleshov7**
### What's done:
1) Fixing detection of companion objects
2) Covering with tests
3) Adding exceptions for 3.4 rule

### CODE REVIEW

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt [hunk 1]
```diff
@@ -181,11 +181,11 @@ class IdentifierNaming : Rule("identifier-naming") {
      */
     private fun checkObjectNaming(node: ASTNode): List<ASTNode> {
         val objectName: ASTNode? = node.getIdentifierName()
-        // checking object naming, the only extension is "companion" keyword
-        if (!(objectName!!.text.isPascalCase()) && objectName.text != "companion") {
+        // if this object is companion object - it does not have any name
```
Comment (petertrr) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:184]: It corresponds to anonymous objects too.

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt [hunk 2]
```diff
@@ -322,8 +322,34 @@ class ASTNodeUtilsTest {
...
+    fun `regression - check for companion object`() {
+       var code = """
+                object Test {
+                    val id = 1
+            	}
+        """.trimIndent()
+
+        applyToCode(code) { node ->
+            if (node.elementType == PROPERTY) {
+                Assert.assertTrue(!node.isNodeFromCompanionObject())
```
Comment (petertrr) [inline diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt:334]:
```suggestion
                Assert.assertFalse(node.isNodeFromCompanionObject())
```

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt (hunk 1 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/IdentifierNaming.kt:184]: yes, true, but this does not change the logic :) I will fix the comment

File: diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/utils/ASTNodeUtilsTest.kt]: for sure :)

**Pull request merged**